pac4j-core/src/main/java/org/pac4j/core/credentials/authenticator/LocalCachingAuthenticator.java

```diff
@@ -56,10 +59,34 @@ public void validate(final T credentials, final WebContext context) throws HttpA
             });
             credentials.setUserProfile(profile);
             logger.debug("Found cached credential. Using cached profile {}...", profile);
-        } catch (Exception e) {
-            throw new CredentialsException(e);
+        } catch (ExecutionException e) {
+            // this covers all the checked exception of validate
+            if (e.getCause() instanceof HttpAction) {
+                throw (HttpAction) e.getCause();
+            } else {
+                throw rethrowWrapper(e);
+            }
+        } catch (UncheckedExecutionException e) {
+            if (e.getCause() instanceof RuntimeException) {
+                throw (RuntimeException) e.getCause();
```

Comment:
One way of simplifying a lot this code is to use `cache.getIfPresent()` and simply populate the cache by hand in case the value is not there: like this there is no need at all to manipulated the various `ExecutionException`, `UncheckedExecutionException` and `ExecutionError`.
We loose potential atomicity of the operation, but I don't think it is a problem functionally, and anyway I don't think the cache are implementing atomicity in practice either.
I can update the PR if you want :)